File: julia/p002.jl
# Even Fibonacci numbers

LIMIT = 4_000_000


function solve()
    a, b = 1, 2
    answer = 0
    while a < LIMIT
        if a % 2 == 0
            answer += a
        end
        a, b = b, a + b
    end
    return answer
end


print(Int(solve()))

File: haskell/p002.hs
-- Even Fibonacci numbers

fib :: Int -> Int -> [Int]
fib a b = a : fib b (a + b)

evenFibonaccis :: [Int]
evenFibonaccis = [x | x <- takeWhile (< 4000000) (fib 1 1), even x]


main :: IO ()
main = do
    print (sum evenFibonaccis)
